File: backend/app/api/v1/schemas/training_status.py
from datetime import date, datetime

from pydantic import BaseModel, Field


class DailyTrainingStatusResponse(BaseModel):
    date: date
    timezone_name: str
    total_load: float
    fitness: float = Field(
        description="Carga crónica suavizada con constante temporal de 42 días."
    )
    fatigue: float = Field(
        description="Carga aguda suavizada con constante temporal de 7 días."
    )
    form: float = Field(description="Fitness menos fatigue.")
    history_day_number: int
    is_warmup: bool = Field(
        description="Indica que todavía no existen 85 días completos de historia."
    )
    training_load_algorithm_version: str
    manual_strength_algorithm_version: str
    training_status_algorithm_version: str
    calculated_at: datetime

File: frontend/src/types/trainingStatus.ts
export const TRAINING_STATUS_PERIODS = [4, 8, 12] as const;
export type TrainingStatusPeriodWeeks = (typeof TRAINING_STATUS_PERIODS)[number];

export interface DailyTrainingStatus {
  date: string;
  timezone_name: string;
  total_load: number;
  fitness: number;
  fatigue: number;
  form: number;
  history_day_number: number;
  is_warmup: boolean;
  training_load_algorithm_version: string;
  manual_strength_algorithm_version: string;
  training_status_algorithm_version: string;
  calculated_at: string;
}

export interface TrainingStatusQuery {
  startDate: string;
  endDate: string;
  timezoneName: string;
  trainingLoadAlgorithmVersion?: string;
  manualStrengthAlgorithmVersion?: string;
  trainingStatusAlgorithmVersion?: string;
}

export type LatestTrainingStatusQuery = Omit<
  TrainingStatusQuery,
  "startDate" | "endDate"
>;
